CodeSniffer/Standards/Generic/Sniffs/PHP/SyntaxSniff.php

```diff
@@ -56,7 +56,30 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
     {
         $phpPath = PHP_CodeSniffer::getConfigData('php_path');
         if ($phpPath === null) {
-            return;
+            // PHP_BINARY is available in PHP 5.4+
+            if (defined('PHP_BINARY')) {
+                $phpPath = PHP_BINARY;
+            }
+            else {
+                // Fallback for old PHP versions that can be removed once PHP
+                // 5.4 is required.
+                $candidates = array(
+                  PHP_BINDIR . '/php',
+                  PHP_BINDIR . '/php.exe',
+                );
+
+                foreach ($candidates as $candidate) {
+                    if (file_exists($candidate) && is_executable($candidate)) {
+                        $phpPath = $candidate;
+                        break;
+                    }
+                }
+
+                if (empty($phpPath)) {
+                    $phpcsFile->addWarning('Unable to check syntax as php_path is not configured', $stackPtr, 'PHPSyntax');
```

Comment:
This is a BC break, because under same conditions before no warning was generated.